test/test_torch.py

```diff
@@ -15954,6 +15954,25 @@ def test_eig(self, device, dtype):
             'A should not contain infs or NaNs',
             lambda: torch.eig(np.nan * torch.ones((2, 2))))
 
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.double)
+    def test_eig_out(self, device, dtype):
+        # the out version of torch.eig needs to be tested manually: we can't
+        # use the "test_out=True" parameter to tensor_op_tests because the
+        # signature is irregular (since we have *two* output vectors)
+        t = _make_tensor((10, 10), dtype=dtype, device=device)
+        evals, evecs = torch.eig(t, eigenvectors=True)
+        #
+        # check that the out= version computes the same values as the normal one
+        out_evals = torch.full_like(evals, fill_value=math.nan)
+        out_evecs = torch.full_like(evecs, fill_value=math.nan)
+        evals2, evecs2 = torch.eig(t, eigenvectors=True, out=(out_evals, out_evecs))
```

Comment:
Add a test of inplace usage of `out`:
```
assert evals2.data_ptr() == out_evals.data_ptr()
assert evecs2.data_ptr() == out_evecs.data_ptr()
```
See copy/inplace discussion above.

Reply:
I'm already checking that `evals2 is out_evals`, so those two lines are redundant. What I could maybe do is to save `old_evals_data_ptr = out_evals.data_ptr()` **before** the call and check that `out_evals.data_ptr() == old_evals_data_ptr`, to ensure that we didn't reallocate the buffer unless we had to?

Reply:
I think just checking that the correct result (same as evals/evecs) is stored in `out_evecs` and that the `data_ptr` stays the same would be good. I think `evals2 is out_evals` might not be as good as `data_ptr` because I'm not sure we have a guarantee that a given C++ Tensor is only wrapped by one PyTorch object (and if we do, we don't need to test).

Reply:
I agree with @t-vi , using `is` operator between pytorch tensors may give false-negative result.

Reply:
fixed by https://github.com/pytorch/pytorch/pull/44105/commits/005a7a55c1627d8b3ccf8fdb6c3c6b81c52d67d8